offer-manager-domain/src/main/groovy/life/qbic/business/offers/Offer.groovy

```diff
@@ -189,6 +211,40 @@ class Offer {
         return overheadSum
     }
 
+
+    /**
+     * This method returns the net cost of all product items for which no overhead cost is calculated
+     *
+     * @return net cost of product items without overhead cost
+     */
+    double getNoOverheadItemsNet() {
+        double costNoOverheadItemsNet = 0
+        for (item in items) {
+            if (item.product instanceof DataStorage || item.product instanceof ProjectManagement) {
+                costNoOverheadItemsNet += item.quantity * item.product.unitPrice
+            }
+        }
+        return costNoOverheadItemsNet
+    }
+
+    /**
+     * This method returns the net cost of product items for which an overhead cost is calculated
+     *
+     * @return net cost of product items with overhead cost
+     */
+    double getOverheadItemsNet() {
+        double costOverheadItemsNet = 0
+        for (item in items) {
+            if (item.product instanceof DataStorage || item.product instanceof ProjectManagement) {
+                // No overheads are assigned for data storage and project management
+            }
```

Comment:
why not doing something like:
if (item.product instanceof Sequencing || item.product instanceof PrimaryBioinformticsAnalysis || item.product instanceof SecondaryBioinformticsAnalysis) ... dunno which types are included here. With this you could delete the empty if statement

Reply:
Nice Idea 🧠